From 96e5a291775c9a0e04bbc5dd7ed8d5a494279470 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 27 Jan 2021 12:43:25 +0000
Subject: [PATCH] tdf#138727 help browser didn't flow text

because of its size request, while the paned shrink lets the pane shrink
below the size request the contents stay at a min of the size request
and aren't really shrunk, just clipped. Instead don't let the pane
shrink but don't set a size.request letting it shrink to its natural
min of its contents which are then truly resized so the text flow works.

Change-Id: Iaf6738ace28c82379ac6322310de9a22f9653473
---
 sfx2/source/appl/newhelp.cxx   | 1 -
 sfx2/uiconfig/ui/helpwindow.ui | 5 ++---
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/sfx2/source/appl/newhelp.cxx b/sfx2/source/appl/newhelp.cxx
index 9160c231f7e3..fb3ed8d5b78b 100644
--- a/sfx2/source/appl/newhelp.cxx
+++ b/sfx2/source/appl/newhelp.cxx
@@ -2467,7 +2467,6 @@ SfxHelpWindow_Impl::SfxHelpWindow_Impl(
     m_xContainer->connect_size_allocate(LINK(this, SfxHelpWindow_Impl, ResizeHdl));
     m_xHelpPaneWindow = m_xBuilder->weld_container("helppanewindow");
     m_xHelpTextWindow = m_xBuilder->weld_container("helptextwindow");
-    m_xHelpTextWindow->set_size_request(m_xHelpTextWindow->get_approximate_digit_width() * 120, -1);
     m_xHelpTextXWindow = m_xHelpTextWindow->CreateChildFrame();
 
     pHelpInterceptor->InitWaiter( this );
diff --git a/sfx2/uiconfig/ui/helpwindow.ui b/sfx2/uiconfig/ui/helpwindow.ui
index 6c9fae044c45..f35acaf06bf7 100644
--- a/sfx2/uiconfig/ui/helpwindow.ui
+++ b/sfx2/uiconfig/ui/helpwindow.ui
@@ -20,7 +20,7 @@
         </child>
       </object>
       <packing>
-        <property name="resize">True</property>
+        <property name="resize">False</property>
         <property name="shrink">False</property>
       </packing>
     </child>
@@ -43,7 +43,6 @@
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
                 <property name="toolbar_style">icons</property>
-                <property name="show_arrow">False</property>
                 <child>
                   <object class="GtkToolButton" id="index">
                     <property name="visible">True</property>
@@ -196,7 +195,7 @@
       </object>
       <packing>
         <property name="resize">True</property>
-        <property name="shrink">True</property>
+        <property name="shrink">False</property>
       </packing>
     </child>
   </object>
-- 
2.28.0

